src/app/api/wallet/search/route.ts

```diff
@@ -0,0 +1,37 @@
+import { getServerSession } from "next-auth";
+import { NextRequest, NextResponse } from "next/server";
+import { options } from "../../auth/[...nextauth]/options";
+import { seachUsernameByWalletAddress } from "../service";
+
+export interface SplitRecipient {
+  username?: string;
+  ens?: string;
+  percentAllocation: number;
+  address: string;
+}
+
+export async function POST(req: NextRequest) {
+  const session = await getServerSession(options);
+  const splitRecipient = (await req.json()) as SplitRecipient[];
+  if (session?.userId && splitRecipient) {
+    const mapedRecipient = await Promise.all(
```

Comment:
mapped*

on another note, I think you should take failure into account and put this around a try/catch, specially due to the reason that this could be updated